wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt

```diff
@@ -0,0 +1,78 @@
+package de.schildbach.wallet.ui.dashpay
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.widget.AppCompatImageView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import de.schildbach.wallet_test.R
+import kotlinx.android.synthetic.main.activity_crop_image.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.dash.wallet.common.InteractionAwareActivity
+import java.io.File
+
+class CropImageActivity : InteractionAwareActivity() {
+
+    companion object {
+        private const val TEMP_FILE = "temp_file"
+        private const val DESTINATION_FILE = "destination_file"
+
+        fun createIntent(context: Context, tempFile: File, destinationFile: File): Intent {
+            val intent = Intent(context, CropImageActivity::class.java)
+            intent.putExtra(TEMP_FILE, tempFile)
+            intent.putExtra(DESTINATION_FILE, destinationFile)
```

Comment:
@sambarboza it isn't a good idea to use serializable extras to share files between Activities. First because the Intent extras size is limited (https://www.neotechsoftware.com/blog/android-intent-size-limit) and second because it is slow.
Couldn't we pass only the files URIs?